test/e2e/index.test.ts

```diff
@@ -126,5 +126,63 @@ describe('e2e', () => {
         }),
       ).to.equal('Gucci');
     });
+
+    it('should reorder columns by dropping into the header', async () => {
+      await renderFixture('DataGrid/ColumnReorder');
+      expect(
+        await page.evaluate(() => document.querySelector('[role="row"]')!.textContent!),
+      ).to.equal('brandyear');
+      const brand = await page.$('[role="columnheader"][aria-colindex="1"] > [draggable]');
+      const brandBoundingBox = await brand?.boundingBox();
+      const year = await page.$('[role="columnheader"][aria-colindex="2"] > [draggable]');
+      const yearBoundingBox = await year?.boundingBox();
+      if (brandBoundingBox && yearBoundingBox) {
+        // Based on https://stackoverflow.com/a/64746679/2801714
+        await page.mouse.move(
+          brandBoundingBox.x + brandBoundingBox.width / 2,
+          brandBoundingBox.y + brandBoundingBox.height / 2,
+          { steps: 5 },
+        );
+        await page.mouse.down();
+        await page.mouse.move(
+          yearBoundingBox.x + yearBoundingBox.width / 2,
+          yearBoundingBox.y + yearBoundingBox.height / 2,
+          { steps: 5 },
+        );
+        await page.mouse.up();
+      }
+      expect(
+        await page.evaluate(() => document.querySelector('[role="row"]')!.textContent!),
+      ).to.equal('yearbrand');
+    });
+
+    it('should reorder columns by dropping into the body', async () => {
+      await renderFixture('DataGrid/ColumnReorder');
+      expect(
+        await page.evaluate(() => document.querySelector('[role="row"]')!.textContent!),
+      ).to.equal('brandyear');
+      const brand = await page.$('[role="columnheader"][aria-colindex="1"] > [draggable]');
+      const brandBoundingBox = await brand?.boundingBox();
+      const cell = await page.$('[role="cell"][data-rowindex="0"][data-colindex="1"]');
+      const cellBoundingBox = await cell?.boundingBox();
+      if (brandBoundingBox && cellBoundingBox) {
+        // Based on https://stackoverflow.com/a/64746679/2801714
+        await page.mouse.move(
+          brandBoundingBox.x + brandBoundingBox.width / 2,
+          brandBoundingBox.y + brandBoundingBox.height / 2,
+          { steps: 5 },
+        );
```

Comment:
Looking at the logic, we could almost consider a drag & drop abstraction. Two repetitions might be a bit short to justify it. So far, it seems to show potential.

Reply:
Not just abstract the drag & drop but also other `page.evaluate` in this file.

Reply:
We could have a chat with Sebastian about how to abstract the common patterns. We might be close to have enough tests here and in the core to make good decisions.